plugins/dss/hooks.js

```diff
@@ -15,10 +15,196 @@ import FormGroupHeading from "#SRC/js/components/form/FormGroupHeading";
+        <FieldLabel>
+          <FormGroupHeading>
+            <FormGroupHeadingContent primary={true}>
+              Containers
+            </FormGroupHeadingContent>
+          </FormGroupHeading>
+        </FieldLabel>
+      );
+      pathLabel = (
+        <FieldLabel>
+          <FormGroupHeading>
+            <FormGroupHeadingContent primary={true}>
+              Container Path
+            </FormGroupHeadingContent>
+          </FormGroupHeading>
+        </FieldLabel>
+      );
+    }
+
+    return (
+      <FormRow key={containerIndex}>
+        <FormGroup className="column-3">
+          {containersLabel}
+          <div className="form-control-input-height">
+            {getContainerNameWithIcon(container)}
+          </div>
+        </FormGroup>
+        <FormGroup className="column-9">
+          {pathLabel}
+          <FieldInput
+            name={`volumeMounts.${volumeMountIndex}.mountPath.${containerIndex}`}
+            type="text"
+            value={volumeMounts[volumeMountIndex].mountPath[containerIndex]}
+          />
+        </FormGroup>
+      </FormRow>
+    );
+  });
+}
+
+function VolumeMountTypesSelect(props) {
+  return (
+    <FieldSelect
+      name={`volumeMounts.${props.index}.type`}
+      value={props.volumes.type}
+    >
+      <option value="">Select...</option>
+      <option value={VolumeConstants.type.host}>Host Volume</option>
+      <option value={VolumeConstants.type.ephemeral}>
+        Ephemeral Volume
+      </option>
+      <option value={VolumeConstants.type.localPersistent}>
+        Local Persistent Volume
+      </option>
+      <option value={VolumeConstants.type.dss}>
+        DC/OS Storage Service
+      </option>
+    </FieldSelect>
+  );
+}
+
+function DSSInput(props) {
+  const { volumes, index, offset } = props;
+  if (volumes.type !== VolumeConstants.type.dss) {
+    return (
+      <div>
+        <FieldLabel>
+          Unable to edit this Volume
+        </FieldLabel>
+        <pre>
+          {JSON.stringify(omit(volumes, ["type", "mountPath"]), null, 2)}
+        </pre>
+      </div>
+    );
+  }
+
+  const nameError = errorsLens
+    .at(index + offset, {})
+    .attr("volumes", {})
+    .get(props.errors).name;
+
+  return (
+    <div>
+      <FormRow>
+        <FormGroup className="column-6" showError={false}>
+          <FieldLabel>
+            <FormGroupHeading>
+              <FormGroupHeadingContent primary={true}>
+                Volume Type
+              </FormGroupHeadingContent>
+            </FormGroupHeading>
+          </FieldLabel>
+          <MountService.Mount
+            type="CreateService:MultiContainerVolumes:Types"
+            volumes={volumes}
+            index={index}
+          >
+            <FieldSelect
+              name={`volumeMounts.${index}.type`}
+              value={volumes.type}
+            >
+              <option value="">Select...</option>
+              <option value={VolumeConstants.type.host}>Host Volume</option>
```

Comment:
Would be nice to use these constants in the VolumesFormSection: https://github.com/dcos/dcos-ui/blob/master/plugins/services/src/js/components/forms/VolumesFormSection.js#L350